gateway-js/src/executeQueryPlan.ts

```diff
@@ -261,36 +261,25 @@ export function shouldSkipFetchNode(
   node: FetchNode,
   variables: VariableValues = {},
 ) {
-  if (node.inclusionConditions) {
-    const shouldSkip = node.inclusionConditions.every((conditionals) => {
-      function resolveConditionalValue(conditional: 'skip' | 'include') {
-        const conditionalType = typeof conditionals[conditional];
-        if (conditionalType === 'boolean') {
-          return conditionals[conditional] as boolean;
-        } else if (conditionalType === 'string') {
-          return variables[conditionals[conditional] as string] as string;
-        } else {
-          return null;
-        }
-      }
-
-      const includeValue = resolveConditionalValue('include');
-      const skipValue = resolveConditionalValue('skip');
-
-      if (includeValue !== null && skipValue !== null) {
-        return !includeValue || skipValue;
-      } else if (includeValue !== null) {
-        return !includeValue;
-      } else if (skipValue !== null) {
-        return skipValue;
+  if (!node.inclusionConditions) return false;
+
+  return node.inclusionConditions.every((conditionals) => {
+    function resolveConditionalValue(conditional: 'skip' | 'include') {
+      const conditionalType = typeof conditionals[conditional];
+      if (conditionalType === 'boolean') {
+        return conditionals[conditional] as boolean;
+      } else if (conditionalType === 'string') {
+        return variables[conditionals[conditional] as string] as boolean;
```

Comment:
The diff is hard to read so it's worth noting that I caught an error here (was casting `as string` when I should've been casting `as boolean`.